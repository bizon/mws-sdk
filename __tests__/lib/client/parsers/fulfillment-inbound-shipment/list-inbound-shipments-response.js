const parseXml = require('../../../../../lib/client/parsers')
const parseListInboundShipmentsResponse = require('../../../../../lib/client/parsers/fulfillment-inbound-shipment/list-inbound-shipments-response')

describe('lib.client.parsers.fulfillment-inbound-shipment.list-inbound-shipments-response', () => {
  it('should parse the ListInboundShipments example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <ListInboundShipmentsResponse xmlns="http://mws.amazonaws.com/FulfillmentInboundShipment/2010-10-01/">
        <ListInboundShipmentsResult>
          <ShipmentData>
            <member>
              <ShipFromAddress>
                <PostalCode>V5V 1A1</PostalCode>
                <Name>jsowprni Devo CA20</Name>
                <CountryCode>CA</CountryCode>
                <StateOrProvinceCode>BC</StateOrProvinceCode>
                <AddressLine1>Address Line 1</AddressLine1>
                <City>Vancouver</City>
              </ShipFromAddress>
              <ShipmentId>FBAN4QNH</ShipmentId>
              <ShipmentName>FBA (8/27/12 1:55 PM)</ShipmentName>
              <ShipmentStatus>WORKING</ShipmentStatus>
              <LabelPrepType>NO_LABEL</LabelPrepType>
              <DestinationFulfillmentCenterId>YYZ1</DestinationFulfillmentCenterId>
              <AreCasesRequired>true</AreCasesRequired>
            </member>
            <member>
              <ShipFromAddress>
                <PostalCode>V5V 1A1</PostalCode>
                <Name>Janani Arvind FBA QA</Name>
                <CountryCode>CA</CountryCode>
                <StateOrProvinceCode>BC</StateOrProvinceCode>
                <AddressLine1>Address 1</AddressLine1>
                <City>Vancouver</City>
              </ShipFromAddress>
              <ShipmentId>FBA1123</ShipmentId>
              <ShipmentName>Test MWS CA Shipment 1</ShipmentName>
              <ShipmentStatus>WORKING</ShipmentStatus>
              <LabelPrepType>NO_LABEL</LabelPrepType>
              <DestinationFulfillmentCenterId>RIC2</DestinationFulfillmentCenterId>
              <BoxContentsSource>NONE</BoxContentsSource>
              <EstimatedBoxContentsFee>
                <TotalUnits>10</TotalUnits>
                <FeePerUnit>
                  <CurrencyCode>USD</CurrencyCode>
                  <Value>0.10</Value>
                </FeePerUnit>
                <TotalFee>
                  <CurrencyCode>USD</CurrencyCode>
                  <Value>10.0</Value>
                </TotalFee>
              </EstimatedBoxContentsFee>
            </member>
          </ShipmentData>
        </ListInboundShipmentsResult>
      </ListInboundShipmentsResponse>`,
    )

    const response = parseListInboundShipmentsResponse(
      '/fulfillmentInboundShipment:ListInboundShipmentsResponse',
      doc,
    )

    expect(response).toMatchSnapshot()
  })
})
