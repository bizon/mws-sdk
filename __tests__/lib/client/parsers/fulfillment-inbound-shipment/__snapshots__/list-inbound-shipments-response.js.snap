// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.fulfillment-inbound-shipment.list-inbound-shipments-response should parse the ListInboundShipments example response from MWS doc 1`] = `
Object {
  "listInboundShipmentsResult": Object {
    "nextToken": null,
    "shipmentData": Array [
      Object {
        "areCasesRequired": true,
        "boxContentsSource": null,
        "confirmedNeedByDate": null,
        "destinationFulfillmentCenterId": "YYZ1",
        "estimatedBoxContentsFee": null,
        "labelPrepType": "NO_LABEL",
        "shipFromAddress": Object {
          "addressLine1": "Address Line 1",
          "addressLine2": null,
          "city": "Vancouver",
          "countryCode": "CA",
          "districtOrCounty": null,
          "name": "jsowprni Devo CA20",
          "postalCode": "V5V 1A1",
          "stateOrProvinceCode": "BC",
        },
        "shipmentId": "FBAN4QNH",
        "shipmentName": "FBA (8/27/12 1:55 PM)",
        "shipmentStatus": "WORKING",
      },
      Object {
        "areCasesRequired": null,
        "boxContentsSource": "NONE",
        "confirmedNeedByDate": null,
        "destinationFulfillmentCenterId": "RIC2",
        "estimatedBoxContentsFee": Object {
          "feePerUnit": Object {
            "currencyCode": "USD",
            "value": 0.1,
          },
          "totalFee": Object {
            "currencyCode": "USD",
            "value": 10,
          },
          "totalUnits": 10,
        },
        "labelPrepType": "NO_LABEL",
        "shipFromAddress": Object {
          "addressLine1": "Address 1",
          "addressLine2": null,
          "city": "Vancouver",
          "countryCode": "CA",
          "districtOrCounty": null,
          "name": "Janani Arvind FBA QA",
          "postalCode": "V5V 1A1",
          "stateOrProvinceCode": "BC",
        },
        "shipmentId": "FBA1123",
        "shipmentName": "Test MWS CA Shipment 1",
        "shipmentStatus": "WORKING",
      },
    ],
  },
  "responseMetadata": Object {
    "requestId": null,
  },
}
`;
